File: Left View of Binary Tree.java
class Tree
{
    //Function to return list containing elements of left view of binary tree.
    ArrayList<Integer> leftView(Node root)
    {
ArrayList<Integer> list = new ArrayList<>();
    if (root == null) return list;

    Queue<Node> queue = new LinkedList<>();
    queue.add(root);

    while (!queue.isEmpty()) {
        int levelSize = queue.size();
        list.add(queue.peek().data); // Add leftmost node

        for (int i = 0; i < levelSize; i++) {
            Node node = queue.poll();
            if (node.left != null) queue.add(node.left);
            if (node.right != null) queue.add(node.right);
        }
    }
    return list;
    }
}

File: Left View of Binary Tree.cpp
class Solution {
  public:
    vector<int> leftView(Node *root) {
        // code here
        
        vector<int> res;
        recurssion(root,0, res);
        return res;
        
    }
    
    void recurssion(Node *root, int level, vector<int> &res){
        
        if(root==NULL){
            return;
        }
        
        if(level==res.size()){
            res.push_back(root->data);
        }
        
        recurssion(root->left,level+1,res);
        recurssion(root->right,level+1,res);
    }
    
    
};

